### ISSUE TITLE
Add job name check to pytorchjob generator helm tool

### ISSUE THREAD

**Issue by kroc353** [project-codeflare/mlbatch#38]
### Name of Feature or Improvement

TL;DR of what you are requesting...

Kubernetes/OpenShift has character limits for pod, services.  Pytorchjob names have `-master-0` and `-worker-N` appended to the pod and service resource that are automatically created.  When the total pod/svc name exceeds 63 characters, the resource cannot be created and the pytorchjob/appwrapper will fail or requeue.  Can the pytorchjob generator run a pre-check against the job name to ensure that the created pod and svc names are less than 63 characters.

### Description of Problem the Feature Should Solve

Multiple times, a training cluster has been impacted by pytorchjobs with exceedingly long names (combined with odd and problematic reqeueing settings).  This feature should help reduce the potential for this problem.  Cannot be completely eliminated if users don't use the tool :)

### Describe the Solution You Would Like to See

Just like the pytorchjob generator tool produces errors when incorrect key/values are supplied to the user file, add a similar check for the length of a job name:

```
jobName: my-job-name-under-50c
```

### Describe Alternatives You Have Considered

One option, outside of responsibility of this repository, would be a Kyverno Policy that reviews job name at admission and takes action there.  We have used Kyverno on the cluster for similar type policies, and ultimately this type of approach could be combined with this tool's length checker verification if we still see this happening a lot.  For example if users adopt some type of pipeline based job launching mechanism, they'd have to generate their appwrapper.  

### Additional Context

https://ibm-research.slack.com/archives/C03QASR0FU3/p1724426831802929

**Comment by tardieu** [comment]
A 50-character limit is already implemented. See https://github.com/project-codeflare/mlbatch/blob/main/tools/pytorchjob-generator/chart/templates/appwrapper.yaml#L5-L7.
